url3c955055f0_dirkzhang_antlr_expressionEngine_tgz-pJ8-expression_DirkExpTestJ8/src/expression/DirkExpLexer.java

```diff
@@ -0,0 +1,1195 @@
+package expression;
+
+import org.antlr.runtime.*;
+import java.util.Stack;
+import java.util.List;
+import java.util.ArrayList;
+
+public class DirkExpLexer extends Lexer {
+    public static final int LT=13;
+    public static final int RBRACE=35;
+    public static final int MOD=21;
+    public static final int LETTER=36;
+    public static final int LBRACE=34;
+    public static final int LTEQ=14;
+    public static final int Exponent=38;
+    public static final int POWER=22;
+    public static final int NOTEQUALS=12;
+    public static final int EQUALS=11;
+    public static final int DATETIME_LITERAL=28;
+    public static final int NOT=23;
+    public static final int GTEQ=16;
+    public static final int MINUS=18;
+    public static final int Digit=37;
+    public static final int AND=10;
+    public static final int EOF=-1;
+    public static final int MUL=19;
+    public static final int LPAREN=24;
+    public static final int LBRACKET=32;
+    public static final int INDEX=4;
+    public static final int RPAREN=25;
+    public static final int STRING_LITERAL=29;
+    public static final int WS=39;
+    public static final int VARIABLE=7;
+    public static final int BRACKETEXP=8;
+    public static final int COMMA=31;
+    public static final int INTEGER_LITERAL=26;
+    public static final int IDENTIFIER=30;
+    public static final int OR=9;
+    public static final int GT=15;
+    public static final int PLUS=17;
+    public static final int MEMBERACCESS=5;
+    public static final int CALL=6;
+    public static final int DIV=20;
+    public static final int RBRACKET=33;
+    public static final int DECIMAL_LITERAL=27;
+
+    // delegates
+    // delegators
+
+    public DirkExpLexer() {;} 
+    public DirkExpLexer(CharStream input) {
+        this(input, new RecognizerSharedState());
+    }
+    public DirkExpLexer(CharStream input, RecognizerSharedState state) {
+        super(input,state);
+
+    }
+    @Override
+	public String getGrammarFileName() { return "F:\\antlr\\BoncExp.g"; }
+
+    // $ANTLR start "AND"
+    public final void mAND() throws RecognitionException {
+        try {
+            int _type = AND;
+            int _channel = DEFAULT_TOKEN_CHANNEL;
+            // F:\\antlr\\BoncExp.g:115:5: ( 'and' )
+            // F:\\antlr\\BoncExp.g:115:9: 'and'
+            {
+            match("and"); 
+
+
+            }
+
+            state.type = _type;
+            state.channel = _channel;
+        }
+        finally {
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 63:78, 84:99, 105:120, 126:140, 146:160, 166:180, 186:200, 206:220, 226:240, 246:260, 266:280, 334:348, 354:369, 375:389, 395:410, 416:430, 436:450, 456:470, 476:490, 496:510, 516:530.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.